test/Flick/AnimatedCarousalPage.xaml.cs

```diff
@@ -49,6 +57,76 @@ public sealed partial class AnimatedCarousalPage : Page
         public AnimatedCarousalPage()
         {
             this.InitializeComponent();
+
+            ElementCompositionPreview.GetElementVisual(sv).Clip = ElementCompositionPreview.GetScrollViewerManipulationPropertySet(sv).Compositor.CreateInsetClip();
+        }
+
+        private object selectedItem = null;
+
+        public object SelectedItem
+        {
+            get
+            {
+                return selectedItem;
+            }
+            set
+            {
+                selectedItem = value;
+            }
+        }
+
+        private bool isScrolling = false;
+
+        public bool IsScrolling
+        {
+            get
+            {
+                return isScrolling;
+            }
+        }
+
+        private double itemScaleRatio = 0.5;
+
+        public double ItemScaleRatio
+        {
+            get
+            {
+                return itemScaleRatio;
+            }
+            set
+            {
+                itemScaleRatio = value;
+            }
+        }
+
+        protected void OnScrollViewerViewChanged(object sender, ScrollViewerViewChangedEventArgs e)
+        {
+            var svCenterPoint = sv.HorizontalOffset + sv.ViewportWidth / 2;
+            int selectedItemIndex = (int)Math.Floor((svCenterPoint + layout.Spacing / 2) / (layout.Spacing + layout.ItemWidth));
+            selectedItemIndex %= ((System.Collections.Generic.IReadOnlyList<object>)repeater.ItemsSource).Count;
+            var selectedUIElement = repeater.TryGetElement(selectedItemIndex);
+            SelectedItem = (selectedUIElement == null ? null : ((UserControl)selectedUIElement).DataContext);
+
+            if (e.IsIntermediate)
+            {
+                if (!isScrolling)
+                {
+                    isScrolling = true;
+                }
+            }
+            else
+            {
+                textBlock.Text = "Selected Item: " + (SelectedItem == null ? "null" : selectedItemIndex.ToString());
+                isScrolling = false;
+            }
+        }
+
+        protected void OnScrollViewerViewChanging(object sender, ScrollViewerViewChangingEventArgs e)
+        {
+            if (!isScrolling)
+            {
+                isScrolling = true;
+            }
```

Comment:
you either don't need this method or the logic above in e.IsIntermediate. Seems redundant.

Reply:
It's not redundant. It's necessary.